src/components/inputs/date/date.component.tsx

```diff
@@ -97,6 +107,7 @@ const DateField: React.FC<FormFieldProps> = ({ question, onChange, handler, prev
             <div className={styles.datePickerSpacing}>
               <Layer>
                 <OpenmrsDatePicker
+                  key={key}
```

Comment:
Could we use the field id as the key instead?

Reply:
@arodidev When using a dynamic key like `field.value`, the component may not behave as expected in some cases. So using the original approach which also ensure the component updates correctly when unspecified is check works well with tests as well

Reply:
The question id is not dynamic, and would be much more suited rather than the int here

Reply:
The major issue is with how the OpenmrsDatePicker propagates the states from the main component to the DatePickerInput component in esm core, when the value is cleared it never trickles down cc @samuelmale
We can report this bug but doing this as well will also re-render the component thus clearing the input when unspecified is check

Reply:
@ibacher any thoughts on the above?

Reply:
This should be fixed.